pkg/ansible/proxy/cache_response.go

```diff
@@ -141,9 +147,33 @@ func (c *cacheResponseHandler) ServeHTTP(w http.ResponseWriter, req *http.Reques
 }
 
 // skipCacheLookup - determine if we should skip the cache lookup
-func (c *cacheResponseHandler) skipCacheLookup(r *requestfactory.RequestInfo) bool {
-	// check if resource is present on request
-	if !r.IsResourceRequest {
+func (c *cacheResponseHandler) skipCacheLookup(r *requestfactory.RequestInfo, gvk schema.GroupVersionKind, req *http.Request) bool {
+
+	owner, err := getRequestOwnerRef(req)
+	if err != nil {
+		log.Error(err, "Could not get owner reference from proxy.")
+		return false
+	}
+	ownerGV, err := schema.ParseGroupVersion(owner.APIVersion)
+	if err != nil {
+		m := fmt.Sprintf("Could not get group version for: %v.", owner)
+		log.Error(err, m)
+		return false
+	}
+	ownerGVK := schema.GroupVersionKind{
+		Group:   ownerGV.Group,
+		Version: ownerGV.Version,
+		Kind:    owner.Kind,
+	}
+
+	// Make sure that the gvk is not blacklisted
+	relatedController, ok := c.cMap.Get(ownerGVK)
+	if !ok {
+		log.Info("Could not find existing watch for gvk.", "ownerGVK:", ownerGVK)
```

Comment:
Just checking... just because a GVK is in the controller map doesn't mean there's an existing _watch_ right? It just means there's an existing controller for the GVK (which I think we always expect?). If `relatedController.Blacklist` contains the GVK, _then_ there won't be a watch. Do I have that right?

Reply:
Yeah, you're right-- fixed it.